src/main/java/spark/QueryParamsMap.java

```diff
@@ -34,6 +34,11 @@
     private static final QueryParamsMap NULL = new NullQueryParamsMap();
 
     /**
+     * Pattern for parsing the key of querystring
+     */
+    private static final Pattern KEY_PATTERN = Pattern.compile("\\A[\\[\\]]*([^\\[\\]]+)\\]*");
```

Comment:
I'm not the author of this code but as my understanding goes the matcher that's fetched from the pattern instance has an internal state. This change means that all requests would share that state. Doesn't feel right.